### DISCUSSION TITLE
Exclude the user-specified DOM element from dismissalbeElement in Combobox.

### DISCUSSION THREAD

**Discussion by erm1116** [chakra-ui/zag#1588] [Ideas]
# 🚀 Feature request
In Combobox component, we want to exclude the specified DOM element from the dismissableElement.
This works well when combining TagsInput and Combobox.

## 🧱 Problem Statement / Justification
The combination with TagsInput is useful when building Combobox(multiple).
(I also know that there is an example in composition: https://github.com/chakra-ui/zag/blob/main/examples/next-ts/pages/composition/combo-tags.tsx )

At this time, we want to provide consistency in the behaviour of Combobox.
Specifically, we want to prevent the Combobox content from closing when a Tag's closeTrigger is clicked. 
In other words, we want to prevent it from being dismissed.

current behavior is like below:

- When clicking on an item in the listbox that is still selected: the listbox remains open

https://github.com/chakra-ui/zag/assets/38889285/6a08c0ea-850d-4686-88fc-c90f43bb012c

- Deleting a tag in input with Backspace: listbox remains open

https://github.com/chakra-ui/zag/assets/38889285/152690fd-c9b1-44d7-bbd2-e3c983f66673

- When clicking on the clear trigger of a tag: the listbox closes

https://github.com/chakra-ui/zag/assets/38889285/1e1c34c8-d2b6-490e-9306-03b0d13703f3

## ✅ Proposed solution or API
Allowing the user to specify the exclude target DOM of the `trackDismissableElement` may solve this problem.
For example, I am considering adding a context such as `excludeDismissableElements` and passing it to the exclude of `trackDismissableElement`. 
This would allow for preventing dismiss on certain elements in addition to the existing `input`, `trigger`, and `clearTrigger`.

But I'm not sure if the idea is appropriate, so I'd like your opinion.

**Comment by segunadebayo** [comment]
Love the detailed breakdown of your request. Makes it easier to chime in.

There's no need to introduce a new property in this case. You can leverage the `onInteractionOutside` event of the combobox to prevent closing when clicking on the tags.

```jsx
function onInteractOutside(event) {
          const { target } = event.detail.originalEvent
          if (!isHTMLElement(target)) return
          if (target.closest("[data-part=tag]")) {
            event.preventDefault
          }
   },
```

That goes into the machine like so:

![CleanShot 2024-06-14 at 09 17 10@2x](https://github.com/chakra-ui/zag/assets/6916170/e4667dfb-3979-4371-a19c-950dc2c2d201)

**Reply by erm1116** [reply]
@segunadebayo 
You're right!!

https://github.com/chakra-ui/zag/blob/5c1ff93c2039d81f9ae7b321701bd52163c26867/packages/utilities/dismissable/src/dismissable-layer.ts#L69-L79

It is checking defaultPrevented in the above section, so the only thing I have to do is calling `event.preventDefault()` conditionally.
Thanks for the quick response!

**Reply by segunadebayo** [reply]
Yes, calling `event.preventDefault()` is all you have to do.
